screens/BooksScreen.js

```diff
@@ -21,28 +21,69 @@ export default function BooksScreen() {
     getBooks();
   }, []);
 
-  const authorName = data.map((e) => {
-    e.authors.map((author) => {
-      console.log(author.name)
-      return author.name
-    })
-     });
+  //looking for ISBN
+  const availabilities =  data.map((book) => {
+    return !!book && book.availability
+  });
+
+  const isbnArray = availabilities.map((availability)=>{
+    console.log("availability.isbn:", availability !== undefined ? availability.isbn : '')
+    return availability !== undefined ? availability.isbn : ''
+  })
+    console.log("isbnArray:", isbnArray)
+
+  const isbn = isbnArray.forEach((isbn) => {
+    console.log("ISBN:",  typeof isbn === "string" ? isbn : '')
+    return typeof isbn === "string" ? `[ISBN: ${isbn}]` : ''
+  })
+  console.log("constISBN:", isbn)
```

Comment:
output: <img width="200" alt="Screenshot 2021-11-01 at 17 23 07" src="https://user-images.githubusercontent.com/43641321/139706990-f42a2d34-fec3-4d29-8296-4dbea1583e52.png">

Reply:
[Array.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) vs [Array.foreach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/foreach)

This should point you in the right direction to understand what is currently going on.

We discussed the difference last time we spoke.

Reply:
finally solved in 02bf2fed85b876743338afb291853980ce0fcc3f. 
There is no need to loop through data to get isbn, meaning I'm not using neither .map or .forEach